Packs/SymantecEndpointSecurity/Integrations/SymantecEndpointSecurity/SymantecEndpointSecurity.py

```diff
@@ -0,0 +1,459 @@
+    def __init__(
+        self,
+        base_url: str,
+        token: str,
+        stream_id: str,
+        channel_id: str,
+        verify: bool,
+        proxy: bool,
+        fetch_interval: int,
+    ) -> None:
+
+        self.headers: dict[str, str] = {}
+        self.token = token
+        self.stream_id = stream_id
+        self.channel_id = channel_id
+        self.fetch_interval = fetch_interval
+
+        super().__init__(
+            base_url=base_url,
+            verify=verify,
+            proxy=proxy,
+            timeout=180,
+        )
+
+        self.get_token()
+
+    def get_token(self):
+        """
+        Retrieves an access token using the `token` provided in the params.
+        """
+        get_token_headers: dict[str, str] = {
+            "accept": "application/json",
+            "content-type": "application/x-www-form-urlencoded",
+            "Authorization": f"Basic {self.token}",
+        }
+        try:
+            res = self._http_request(
+                "POST",
+                url_suffix="/v1/oauth2/tokens",
+                headers=get_token_headers,
+                data={},
+            )
+        except Exception as e:
+            demisto.info(f"Failure in get_token method, Error: {e}")
+            raise e
+        try:
+            self.headers = {
+                "Authorization": f'Bearer {res["access_token"]}',
+                "Accept": "application/x-ndjson",
+                "Content-Type": "application/json",
+                "Accept-Encoding": "gzip",
+            }
+        except KeyError:
+            raise DemistoException(
+                f"The key 'access_token' does not exist in response, Response from API: {res}"
+            )
+
+    def get_events(self, payload: dict[str, str]):
+        """
+        API call in streaming to fetch events
+        """
+        return self._http_request(
+            method="POST",
+            url_suffix=f"/v1/event-export/stream/{self.stream_id}/{self.channel_id}",
+            json_data=payload,
+            params={"connectionTimeout": DEFAULT_CONNECTION_TIMEOUT},
+            resp_type="text",
+            headers=self.headers,
+        )
+
+
+def manage_fetch_interval(client, start_run, end_run):
+    """
+    Manages the fetch interval by sleeping if necessary.
+
+    This function calculates the fetch runtime against client.fetch_interval.
+    If the runtime is less than the client.fetch_interval time, it will sleep
+    for the time difference between client.fetch_interval and the fetch runtime.
+    Otherwise, the next fetch will occur immediately.
+
+    Args:
+        client: The client object containing the fetch_interval attribute.
+        start_run: The start time of the fetch operation.
+        end_run: The end time of the fetch operation.
+    """
+    fetch_sleep = client.fetch_interval - (end_run - start_run)
+    if fetch_sleep > 0:
+        time.sleep(fetch_sleep)
+
+
+def normalize_date_format(date_str: str) -> str:
```

Comment:
We use this method lots of times. 
Consider only using it on ingested events, and then we only have one format all around.